src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -93,6 +109,20 @@ public AssessmentDto postAssessment(AssessmentDto assessmentDto, long treatmentI
         return toDto(assessmentSaved, false, false, false, false);
     }
 
+    private AssessmentDto toDto(Assessment assessment, Participant participant, boolean submissions) throws AssessmentNotMatchingException {
+        AssessmentDto assessmentDto = toDto(assessment, null, false, false, submissions, false);
+
+        RetakeDetails retakeDetails = new RetakeDetails();
+        retakeDetails.setKeptScore(submissionService.getScoreFromMultipleSubmissions(participant, assessment));
+
+        List<Submission> participantSubmissions = submissionService.findByParticipantId(participant.getParticipantId());
+        retakeDetails.setSubmissionAttemptsCount(participantSubmissions.size());
```

Comment:
This should only include submissions where dateSubmitted != null.

Reply:
Good catch! I'll fix.